src/Features/Core/Portable/GenerateMember/GenerateVariable/AbstractGenerateVariableService.State.cs

```diff
@@ -151,7 +151,7 @@ internal bool CanGenerateParameter()
                 // !this.IsInMemberContext prevents us offering this fix for `x.goo` where `goo` does not exist
                 // Workaround: The compiler returns IsImplicitlyDeclared = false for <Main>$.
                 return ContainingMethod is { IsImplicitlyDeclared: false, Name: not WellKnownMemberNames.TopLevelStatementsEntryPointMethodName }
-                    && !IsInMemberContext && !IsConstant;
+                    && !IsInMemberContext && !IsConstant && !ContainingMethod.IsAnonymousFunction();
```

Comment:
not exactly what i woudl expect.  i woudl still think we'd offer to add the parameter, but just inside AssertSomething.

Reply:
By "inside" do you mean add the parameter to the lambda, or add the parameter to the AssertSomething method?

Reply:
Took a punt you meant the latter

Reply:
yup, the latter.